modules/tensor_mechanics/include/actions/TensorMechanicsAction.h

```diff
@@ -32,6 +32,9 @@ class TensorMechanicsAction : public TensorMechanicsActionBase
   virtual std::string getKernelType();
   virtual InputParameters getKernelParameters(std::string type);
 
+  template <typename T, typename T2>
```

Comment:
A doxygen comment describing what the parameters are and what this function does would be helpful here.